master/internal/rm/agent_resource_manager.go

```diff
@@ -113,27 +136,54 @@ func (a AgentResourceManager) ResolveResourcePool(
 		req := sproto.GetDefaultComputeResourcePoolRequest{}
 		resp, err := a.GetDefaultComputeResourcePool(ctx, req)
 		if err != nil {
-			return "", fmt.Errorf("defaulting to compute pool: %w", err)
+			return name, fmt.Errorf("defaulting to compute pool: %w", err)
 		}
 		return resp.PoolName, nil
 	}
 
 	if err := a.ValidateResourcePool(ctx, name); err != nil {
-		return "", fmt.Errorf("validating pool: %w", err)
+		return name, fmt.Errorf("validating pool: %w", err)
 	}
+	return name, nil
+}
 
+// ValidateResources ensures enough resources are available for a command.
+func (a AgentResourceManager) ValidateResources(
+	ctx actor.Messenger, name string, slots int, command bool,
+) error {
 	if slots > 0 && command {
-		switch resp, err := a.ValidateCommandResources(ctx, sproto.ValidateCommandResourcesRequest{
-			ResourcePool: name,
-			Slots:        slots,
-		}); {
+		switch resp, err := a.ValidateCommandResources(ctx,
+			sproto.ValidateCommandResourcesRequest{
+				ResourcePool: name,
+				Slots:        slots,
+			}); {
 		case err != nil:
-			return "", fmt.Errorf("validating request for (%s, %d): %w", name, slots, err)
+			return fmt.Errorf("validating request for (%s, %d): %w", name, slots, err)
 		case !resp.Fulfillable:
-			return "", errors.New("request unfulfillable, please try requesting less slots")
+			return errors.New("request unfulfillable, please try requesting less slots")
 		}
 	}
-	return name, nil
+	return nil
+}
+
+// GetResourcePoolAvailability is a default implementation to satisfy the interface.
+func (a AgentResourceManager) GetResourcePoolAvailability(ctx actor.Messenger,
+	name string, slots int) (
+	[]command.LaunchWarning,
+	error,
+) {
+	launchWarnings := []command.LaunchWarning{}
+	if slots == 0 {
+		return launchWarnings, nil
```

Comment:
maybe just this, again? not going to comment this anymore.. i think i've highlighted the pattern. if im missing something lmk.
```suggestion
		return nil, nil
```